src/serverenvironment.cpp

```diff
@@ -2217,6 +2221,8 @@ void ServerEnvironment::deactivateFarObjects(bool _force_delete)
 		obj->removingFromEnvironment();
 		// Deregister in scripting api
 		m_script->removeObjectReference(obj);
+		// stop attached sounds
+		m_server->stopAttachedSounds(id);
```

Comment:
```c++
// Tell the object about removal
obj->removingFromEnvironment();
// Deregister in scripting api
m_script->removeObjectReference(obj);
// stop attached sounds
m_server->stopAttachedSounds(id);
```

Appears 3 times. It would be good to extract it into a function.